### PR TITLE
PUBDEV-5204: Allow user to specify explicit list of interactions in GLM

### PR DESCRIPTION
**Pull request by michalkurka**

### CODE REVIEW

**Comment by michalkurka** [comment]
@h2o-ops please test!

**Comment by michalkurka** [comment]
@h2o-ops please test!

**Comment by michalkurka** [comment]
@mmalohlava can you please do an initial review?

File: h2o-core/src/main/java/hex/StringPair.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package hex;
+
+import water.Iced;
+
+public class StringPair extends Iced {
```
Comment (mmalohlava) [inline h2o-core/src/main/java/hex/StringPair.java]: Minor; self annotation in `Iced<...>` is missing.

File: h2o-algos/src/main/java/hex/schemas/GLMV3.java [hunk 2]
```diff
@@ -164,6 +166,9 @@
     @API(help="A list of predictor column indices to interact. All pairwise combinations will be computed for the list.", direction=Direction.INPUT, level=Level.expert)
     public String[] interactions;
 
+    @API(help="A list of pairwise (first order) column interactions.", direction=Direction.INPUT, level=Level.expert)
```
[the remark this answers is not in the document]
Comment (michalkurka) [reply h2o-algos/src/main/java/hex/schemas/GLMV3.java]: Interactions between 2 variables, pairwise interactions are sometimes called first-order interactions

see eg.: http://www.mdpi.com/2076-0760/2/2/91/pdf-vor

File: h2o-algos/src/main/java/hex/DataInfo.java [hunk 3]
```diff
@@ -331,11 +320,12 @@ public DataInfo(Frame train, Frame valid, int nResponses, boolean useAllFactorLe
...
     DataInfo res = new DataInfo(_adaptedFrame,null,1,_useAllFactorLevels,TransformType.NONE,TransformType.NONE,_skipMissing,_imputeMissing,!(_skipMissing || _imputeMissing),_weights,_offset,_fold);
     res._interactions=_interactions;
-    res._interactionColumns=_interactionColumns;
-    if( _interactionColumns!=null ) {
-      int[] interactions = new int[_interactionColumns.length];
-      for(int i=0;i<interactions.length;++i)
-        interactions[i] = valid.find(_interactionColumns[i]);
+    res._interactionSpec=_interactionSpec;
+    if( _interactionSpec!=null ) {
+      //FIXME: this looks wrong:
```
[the remark this answers is not in the document]
Comment (michalkurka) [reply h2o-algos/src/main/java/hex/DataInfo.java]: Yes, this still needs to be fixed. I will make a unit test that will actually test the case when validation set is provided.

**Comment by michalkurka** [comment]
@h2o-ops please test!

**Pull request merged**